assignments/this.js

```diff
@@ -13,14 +13,75 @@
 
 // code example for Window Binding
 
+
+var sayAge = function(){
+    // 'use strict';
+    console.log(this.age);
+};
+
+var me = {
+    age: 26
+};
+
+sayAge();
+window.age = 2
+sayAge();
+
 // Principle 2
 
 // code example for Implicit Binding
 
+let Person = function(name, state){
+    return{
+        name: name,
+        state: state,
+        introduce: function(){
+            console.log(this.name + ' ' + this.state)
+        },
+        friend: {
+            name: 'Jenn',
+            state: 'Korea',
+            introduce: function(){
+                console.log(this.name + ' ' + this.state)
+            }
+        }
+    }
+}
+
+
+var Jason = Person('Jason', 'California');
+Jason.introduce();
+Jason.friend.introduce();
+
 // Principle 3
 
 // code example for New Binding
 
+var Animal = function(color, name, type){
```

Comment:
I hope you don't mind. I played with the function so it ran: 
var Animal = function(color, name, type){
    this.color = color;
    this.name = name;
    this.type = type;
    this.bark = function() {
      console.log(`My dog's name is ${this.name}. He is ${this.color} and the breed is ${this.type}`)
  }
  }


var dog = new Animal ('black', 'Cherry', 'Schnauzer');

dog.bark()